django/db/models/functions/comparison.py

```diff
@@ -175,21 +179,28 @@ def join(self, args):
         )
 
     def as_postgresql(self, compiler, connection, **extra_context):
-        if not connection.features.is_postgresql_16:
-            copy = self.copy()
-            copy.set_source_expressions(
-                [
-                    Cast(expression, TextField()) if index % 2 == 0 else expression
-                    for index, expression in enumerate(copy.get_source_expressions())
-                ]
-            )
-            return super(JSONObject, copy).as_sql(
+        copy = self.copy()
+        copy.set_source_expressions(
+            [
+                Cast(expression, TextField()) if index % 2 == 0 else expression
+                for index, expression in enumerate(copy.get_source_expressions())
+            ]
+        )
+
+        if connection.features.is_postgresql_16:
+            return copy.as_native(
                 compiler,
                 connection,
-                function="JSONB_BUILD_OBJECT",
+                returning="JSONB",
                 **extra_context,
             )
```

Comment:
@sarahboyce You can see we're still using the native JSONB function on postgres 16. That's not reverted.